Gat/mrd pipeline fixes

GatKrieger opened the pull request:
still work in progress, but would love to have a review.

GatKrieger committed:
parent 649cf08d0110cf9a8f8c1f6e970fffc6e651626c
author GatKrieger <gat.krieger@ultimagen.com> 1698217912 +0000
committer Itai Rusinek <itai.rusinek@ultimagen.com> 1705236770 +0000

remove index equality test

save tumor fraction tables in an h5 file

snvq as tf denominator in case of filtering by qual

fix QUAL to qual

fill in zero supporting reads

fill in zeros in the matched signature plot

fill in coverage for singatures with zero supporting reads

fillin zero in missing motifs (good for very small signatures...)

PR

fix qaul regex

remove AF filtering from MRD analysis

remove index equality test

fill in zero supporting reads

fill in coverage for singatures with zero supporting reads

fillin zero in missing motifs (good for very small signatures...)

PR

fix qaul regex

remove AF filtering from MRD analysis

generate_synthetic_signatures can now take a reference fasta to extract the reference motifs with rather than requiring that X_LM and X_RM exist in the vcf

Added a sorter_stats_to_mean_coverage function to allow a Cromwell task to read the coverage from a Sorter json stats file

mrd analysis can now run without a matched signature

correct use of ref genome in generating db_controls

Added sorter_stats_to_mean_coverage to script list

kenissur commented on ugvc/reports/mrd_automatic_data_analysis.ipynb:
```diff
@@ -279,6 +281,17 @@
+    "        ]\n",
+    "        for mut in all_muts:\n",
+    "            if mut not in x.index:\n",
+    "                x.loc[mut] = 0\n",
```
I think x.reindex(all_muts).fillna(0).astype(int) would save you the loop.

kenissur commented on ugvc/mrd/mrd_utils.py:
```diff
@@ -176,11 +176,6 @@ def collect_coverage_per_locus_gatk(coverage_csv, df_sig):
-        raise ValueError(
-            f"df_sig index names {df_sig.index.names} \
-            do not match coverage_gatk index names {coverage_gatk.index.names}"
-        )
```
Why do you want to remove this check?

kenissur commented on ugvc/reports/mrd_automatic_data_analysis.ipynb:
```diff
@@ -587,13 +577,20 @@
     "    )\n",
     "    df_tf = df_supporting_reads.join(df_coverage).fillna(0)\n",
     "    df_tf[\"corrected_denom\"] = df_tf[\"coverage\"] * denom_ratio\n",
+    "    df_tf[\"corrected_denom\"] = np.ceil(df_tf[\"corrected_denom\"])\n",
```
1. Best to do this together with the line above
2. Why round this value?

kenissur commented on ugvc/reports/mrd_automatic_data_analysis.ipynb:
```diff
@@ -716,31 +751,29 @@
-    "df_supporting_reads_per_locus_filt.to_csv(os.path.join(output_dir, basename + \".df_supporting_reads_per_locus_filt_signature_filt_featuremap.csv\"))\n",
-    "df_supporting_reads_per_locus_unfilt.to_csv(os.path.join(output_dir, basename +  \".df_supporting_reads_per_locus_unfilt_signature_filt_featuremap.csv\"))\n",
-    "df_supporting_reads_per_locus_unfilt2.to_csv(os.path.join(output_dir, basename + \".df_supporting_reads_per_locus_filt_signature_unfilt_featuremap.csv\"))"
+    "## Write tf tables to csv files\n",
```
* to an hdf file

kenissur reviewed (approved):
Just some minor comments

kenissur committed:
SRSNV fixes:
1. in create_report_plots now assert the all the expected params keys exist
2. update test functions interface to change in prepare_featuremap_for_model

GatKrieger committed:
bring mrd analysis notebook from backup

kenissur committed:
Now saving a vector of qualities corresponding to the test X and y data

kenissur committed:
Better implementation of qual_test calculation

kenissur committed:
bugfix

kenissur committed:
1. qual_test fixed - is now interpolated quality rather than ML_QUAL
2. create_report now takes data instead of files, saves intermediate save and load in training. Test adapted.

kenissur committed:
bugfix

kenissur committed:
Plot updates - Showing SNVQ instead of residual error rate, fixed per qual histograms

kenissur committed:
bugfix

GatKrieger committed:
correct denominator using the qual_test dataframe from SRSNV

kenissur committed:
Fixed change in normalization + Inf quality issue

GatKrieger committed:
bug fix in notebook

kenissur commented:
@doron-st and @ilyasoifer, this is a major PR that updates the MRD pipeline. I would appreciate review on whatever you feel comfortable reviewing before we go ahead and merge.

ilyasoifer commented on ugvc/pipelines/mrd/sorter_stats_to_mean_coverage.py:
```diff
@@ -0,0 +1,53 @@
+        Path to output file.
+
+    """
+    mean_cvg = np.ceil(read_effective_coverage_from_sorter_json(sorter_stats_json)[0])
```
Could you give some clarification what effective_coverage means? Either here or in `read_effective_coverage_from_sorter_json`

kenissur replied:
In this case it's just the mean coverage, fixed, and clarified in the other function

ilyasoifer reviewed (approved):
Looks good, just minor comment.

kenissur committed:
PR comment

GatKrieger committed:
no need in srsnv_stats_h5
